### PR TITLE
feat: align FormCard with detroit updates

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

#56

## Description

Aligns the FormCard component with the changes in Detroit. There are no existing stories but you can check out [two new ones](https://deploy-preview-43--storybook-bloom-dev.netlify.app/?path=/story/blocks-form-card--default-card). It's sort of funky that the no header state has no internal styling at all, but that's the existing behavior.

The `isVisible` prop wasn't used anywhere so I removed it since I already had to make a breaking change to that prop to accommodate a custom header priority.

Commit message indicates: `BREAKING CHANGE: removes the isVisible prop and re-types the heading prop`

## How Can This Be Tested/Reviewed?

There weren't any existing stories for this component. If you link you can compare the two versions (title/no title) by looking at an application section (title flavor) and at the forgot password page (no title flavor).

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have made any corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added or updated stories if new changes are not captured in Storybook
- [x] New and existing unit tests pass locally with my changes
- [ ] I have exported any new pieces added to ui-components
- [ ] I have documented this change in the changelog, and any breaking changes are well described

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Either review the Storybook preview or pull the changes down locally and test that the acceptance criteria is met
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

On merge, squash commits.

### CODE REVIEW

File: src/blocks/FormCard.scss [hunk 1]
```diff
@@ -87,11 +88,16 @@
 }
 
 .form-card__group {
-  @apply px-8;
+  @screen md {
```
Comment (emilyjablonski) [inline src/blocks/FormCard.scss:91]: Fixes a bug where we were showing too much space on the bottom on mobile views

File: src/blocks/FormCard.tsx [hunk 2]
```diff
@@ -1,27 +1,27 @@
 import * as React from "react"
+import { Heading, HeadingProps } from "../text/Heading"
 import "./FormCard.scss"
 
 export interface FormCardProps {
-  header?: FormCardHeader
+  header?: HeadingProps
```
Comment (emilyjablonski) [inline src/blocks/FormCard.tsx:6]: Allows for custom priority

File: src/blocks/FormCard.scss [hunk 3]
```diff
@@ -87,11 +88,16 @@
...
+    @apply px-8;
+  }
+
+  @screen print {
```
Comment (ludtkemorgan) [inline src/blocks/FormCard.scss:95]: More out of curiosity, but what does "print" mean here?

**Review by ludtkemorgan — APPROVED** [review]
🦁

File: src/blocks/FormCard.scss (hunk 3 above)
Comment (emilyjablonski) [reply src/blocks/FormCard.scss:95]: To me it looks like a holdover that requires cleanup --> seeds, was just worried about messing with it here

**Review by ColinBuyck — APPROVED** [review]
LGTM 👃

**Pull request merged**